### DISCUSSION TITLE
Adding more detailed suggestions

### DISCUSSION THREAD

**Discussion by szabgab** [rust-lang/rust-clippy#12233] [Ideas]
Once in a while I feel that the explanations of the lints could include more suggestions. e.g [print_stdout](https://rust-lang.github.io/rust-clippy/master/index.html#/print_stdout) could include the suggestion to replace the debugging print statement with `log::debug!`

Something like this:

> One solution would be to convert the print-statements into a logging calls. For this you'll need to add the [log](https://crates.io/crates/log) create and one of the loggers and replace `println!()` by `log::debug!()`

I'd be happy to send such extensions whenever  encounter something, assuming the general concept of giving more detailed suggestion is something that is seen as positive.

**Comment by flip1995** [comment]
We are usually careful with suggesting external crates, as we don't want to put the impression, that we favor one crate over another. 

For logging: Why suggest `log` over any of the other [logging crates](https://github.com/rust-unofficial/awesome-rust?tab=readme-ov-file#logging)?

So I wouldn't put this in the lint suggestion, but we could include this information in the lint documentation and either link the awesome-rust page ~~or the [rust-cookbook](https://rust-lang-nursery.github.io/rust-cookbook/development_tools/debugging/config_log.html) which is a bit more official.~~

EDIT: Just saw that the last commit to the cookbook was 3 years ago, so maybe not that

**Comment by szabgab** [comment]
As I now seem to understand there is some explanation that comes with clippy and is printed on the terminal and there is more information on the web site that clippy refers to. I need to understand the separation between the two, but I was thinking about the web site.

Maybe the link should be to the [logging crates](https://github.com/rust-unofficial/awesome-rust?tab=readme-ov-file#logging) as mentioned above or to to the [logging tag on crates.io](https://crates.io/keywords/logging) So:

> One solution would be to convert the print-statements into logging  calls using one of the [logging crates](https://crates.io/keywords/logging).

**Reply by flip1995** [reply]
> explanation that comes with clippy and is printed on the terminal 

With that I meant the normal error message format, that you also get from `rustc`.

However, there is also `clippy-driver --explain clippy::almost_swapped`, that prints the documentation that is also available on the web.

Adding a link to `crates.io` sounds good to me. Great idea! Feel free to open a PR about this.
